pkg/store/change/device/utils.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package device
+
+import (
+	"github.com/onosproject/onos-config/pkg/types/change/device"
+	devicetopo "github.com/onosproject/onos-topo/pkg/northbound/device"
+	"sort"
+	"strings"
+)
+
+// ExtractFullConfig retrieves the full consolidated config for a Configuration
+// This gets the change up to and including the latest
+// Use "nBack" to specify a number of changes back to go
+// If there are not as many changes in the history as nBack nothing is returned
+func ExtractFullConfig(deviceID devicetopo.ID, newChange *device.Change, changeStore Store,
```

Comment:
Would need some thorough unit tests. 

Also I expected this to be more of a static function that takes a set of deviceChanges - did not expect to see the call to List() in it. Also the order of them is *very* important

Reply:
@ray-milkey store-api_test.go were the tests I mentioned that need adaptation to this new method under `utils.go`